### DISCUSSION TITLE
[2023-03-06] Performance Team

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#5827] [Weekly Demo Day]
Here we go! :)

**Comment by JaroslavTulach** [comment]
Improving engine startup performance:

- #5700
     - Also includes generic `Cache` infrastructure
- #5608
- #5763
- #5698

Also working on engine runtime performance:
- #5745
    - #5782
    - #5781

**Comment by kazcw** [comment]
This week I implemented #5722, finishing the highest benefit-per-cost text improvements.

---

Here's an update on the status of frontend CPU work (including my open PRs that will merge today) since my performance survey of 3 weeks ago (#5603):
![image](https://user-images.githubusercontent.com/1047859/223180042-959f371e-3679-4bd9-84b3-90bc328100c1.png)
This is what eliminating low-hanging fruit looks like! Shaving off further seconds will not be so easy.

---

This week I will be implementing the new documentation lexer (#5447).

**Comment by farmaazon** [comment]
Last week, the cached icons in the Component Browser were merged. There is a regression, [where icons are broken on Windows with non-integral scaling](#5807), the fix is in review. For a couple of next weeks, I will switch to fixing issues not related to the performance.
